test/e2e/e2e_test.go

```diff
@@ -690,6 +683,88 @@ func checkMetricsCollection(t *testing.T, promClient prometheusv1.API, namespace
 	}
 }
 
+func runPromQueryForVector(t *testing.T, promClient prometheusv1.API, query string, sampleTime time.Time) (model.Vector, error) {
+	results, warnings, err := promClient.Query(context.Background(), query, sampleTime)
+	if err != nil {
+		t.Errorf("failed to execute prometheus query: %v", err)
+		return model.Vector{}, err
+	}
+	if len(warnings) > 0 {
+		t.Logf("prometheus query emitted warnings: %v", warnings)
+	}
+
+	result, ok := results.(model.Vector)
+	if !ok {
+		t.Fatalf("expecting scalar type result, found: %v ", reflect.TypeOf(results))
+	}
+
+	return result, nil
+}
+
+func getSampleForPromQuery(t *testing.T, promClient prometheusv1.API, query string, sampleTime time.Time) (*model.Sample, error) {
+	res, err := runPromQueryForVector(t, promClient, query, sampleTime)
+	if err != nil {
+		t.Errorf("failed to execute prometheus query: %v", err)
+		return nil, err
+	}
+	if len(res) == 0 {
+		return nil, fmt.Errorf("no matching metrics found for query %s", query)
+	}
+	return res[0], nil
+}
+
+func checkServiceCAMetrics(t *testing.T, client *kubernetes.Clientset, promClient prometheusv1.API) {
+	timeout := 60 * time.Second
+
+	secret, err := client.CoreV1().Secrets(serviceCAControllerNamespace).Get(signingKeySecretName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("error retrieving signing key secret: %v", err)
+	}
+	currentCACerts, err := util.PemToCerts(secret.Data[v1.TLSCertKey])
+	if err != nil {
+		t.Fatalf("error unmarshaling %q: %v", v1.TLSCertKey, err)
+	}
+	if len(currentCACerts) == 0 {
+		t.Fatalf("no signing keys found")
+	}
+
+	sampleTime := time.Now()
+	want := currentCACerts[0].NotAfter
+	err = wait.PollImmediate(10*time.Second, timeout, func() (bool, error) {
+		have, err := getSampleForPromQuery(t, promClient, `service_ca_expiry_time`, sampleTime)
+		if err != nil {
+			t.Errorf("failed to get sample value: %v", err)
+			return false, nil
+		}
+
+		if float64(want.Unix()) != float64(have.Value) {
+			t.Fatalf("service ca expiry time mismatch expected %v observed %v", float64(want.UnixNano()), float64(have.Value))
+		}
+
+		have, err = getSampleForPromQuery(t, promClient, `service_ca_time_left`, sampleTime)
```

Comment:
nit: Rather than using `have` twice, please use distinct names for the destination variables (e.g. `rawExpiryTime`, `rawTimeLeft`).

Reply:
updated the code